Document tests for nested folders

nfebe opened the pull request:
Document tests for nested folders

- Add test nest for testing nested folders along side instructions.
- Update some minor stuff in readme somewhat related to this.
- Update upload script to accept files from the newly added `nested` dir in `test-tree/misc`.

Resolves #3

kfogel commented:
@mithuna Since I needed to start working with the test data, I did a rewrite (in Python, as I don't know PHP very well).  I'm requesting review partly to get your feedback and partly to give you an example of some Python code that's solving a problem you are already familiar with.

@Fenn-CS Feel free to review too, at your leisure!  But I won't block on review to start working with this, since I need to upload some test data pretty soon.

Fenn-CS commented on generate-tree.py:
```diff
@@ -0,0 +1,101 @@
+    # the enemy of a perfectly okay yak.
+    for ch in (list(range(1, ord('/')))           # SOH through '.'
+               + list(range(ord(':'), ord('A')))  # ':" through '@'
+               + list(range(ord(':'), ord('A')))  # ':" through '@'
```
I am sure why this set of Unicode characters have to be repeated.

Fenn-CS reviewed (approved):
I saw something that looks like a repeated line! Otherwise, the algorithm is sweet and clean :)

kfogel commented:
Thanks for the catch, @Fenn-CS !  Fixed, rebased, and merging now.

mithuna commented:
Thanks for including me in this PR. So happy to learn new coding techniques  from experts.

kfogel commented:
Well, I don't know if I qualify as an expert, but I'm glad it was useful, @mithuna!

nfebe committed:
Document tests for nested folders

- Add test nest for testing nested folders along side instructions.
- Update some minor stuff in readme somewhat related to this.
- Update upload script to accept files from the newly added `nested` dir in `test-tree/misc`.

Signed-off-by: Fon E. Noel NFEBE <fenn25.fn@gmail.com>

nfebe committed:
Improve readme structure

- Take test case headings one level up.
- Improve notes on testing scope.
- Update/fix notes on nested folders from last commit.

Signed-off-by: Fon E. Noel NFEBE <fenn25.fn@gmail.com>

nfebe commented on upload-test.py:
```diff
@@ -49,21 +50,23 @@ def main():
             if fname[0:3] == "002":
-                out = rclone_upload(os.path.join(MISC_DIR, fname), cli.remote_dir)
+                rclone_upload(os.path.join(MISC_DIR, fname), cli.remote_dir)
+    elif os.path.abspath(cli.directory) == os.path.abspath(NESTED_DIR):
```
I wonder why we have this pattern in the code where we verify the directory before accepting to upload, and when the directory is not known, the script refuses to upload.

However, I have continued to use this pattern, hence the addition of the nested directory path.

nfebe reviewed (commented):
Merging after a self-review pending any updates that @jvasile may want to see.
